### DISCUSSION TITLE
missing ) after formal parameters

### DISCUSSION THREAD

**Discussion by oswetto** [mozilla/rhino#1273] [Q&A]
Hello,
 i use rhino 1.7.14 to parse w3c [testharness.js](https://github.com/mozilla/rhino/files/9824517/testharness.txt) but i've parse error 
 "missing ) after formal parameters". This is my code. Any suggest to fix?
 
Reader reader = new InputStreamReader(in, "utf-8"))
BufferedReader br = new BufferedReader(reader);						
ctx.evaluateReader(scope, br, scriptURI, 1, null);

**Comment by p-bakker** [comment]
Other than someone interested in debugging this for you, i doubt you can expect any response to you question.

I suggest you try to pinpoint where in the file the parser fails and see what's going on there.

Might be that it uses a newer JavaScript feature that the Rhino parser not yet supports

**Comment by rbri** [comment]
This is usual the case if your js code uses rest parameters (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters). This is not supported by rhino as of now.
